tests/parser/types/numbers/test_sqrt.py

```diff
@@ -1,22 +1,37 @@
 from decimal import (
     ROUND_FLOOR,
     Decimal,
+    getcontext,
 )
 
+from eth_tester.exceptions import (
+    TransactionFailed,
+)
+import hypothesis
 import pytest
 
+from vyper.utils import (
+    SizeLimits,
+)
+
+getcontext().prec = 168
```

Comment:
I'm a big fan of using the `decimal.localcontext()` context manager API as it 1) preserves existing global settings upon exit and 2) gives stronger guarantees that you are indeed working with the context you desire within the specific block of code in question.

Reply:
Ah cool, will do :+1:

Reply:
@pipermerriam I tried using localcontext() for theses tests now. But I think it doesn't play well with tests - I'd have to wrap each test that currently just uses `Decimal` with a with statement - which seems like serious boiler plate (for tests) e.g.
```python
def test_sqrt_variable(get_contract_with_gas_estimation):
    code = """
@public
def test(a: decimal) -> decimal:
    return sqrt(a)

@public
def test2() -> decimal:
    a: decimal = 44.001
    return sqrt(a)
    """

    c = get_contract_with_gas_estimation(code)

    with localcontext() as decimal_ctx:
        decimal_ctx.prec = PRECISION
        val = Decimal('33.33')
        assert c.test(val) == decimal_sqrt(val)

        val = Decimal('0.1')
        assert c.test(val) == decimal_sqrt(val)

        assert c.test(Decimal('0.0')) == Decimal('0.0')
        assert c.test2() == decimal_sqrt(Decimal('44.001'))

        with pytest.raises(TransactionFailed):
            c.test(Decimal('-1.1'))

```